### PR TITLE
Make `deposit_event` work with none generic events

### PR DESCRIPTION
**Pull request by bkchr**
`fn deposit_event() = default` will now be used for none generic events
`fn deposit_event<T>() = default` is now for generic events.

Fixes: https://github.com/paritytech/substrate/issues/1275

### CODE REVIEW

File: srml/example/src/lib.rs [hunk 1]
```diff
@@ -98,7 +98,9 @@ decl_module! {
 	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
 		/// Deposit one of this module's events by using the default implementation.
 		/// It is also possible to provide a custom implementation.
-		fn deposit_event() = default;
+		/// For none generic events, the generic parameter just needs to be dropped, so that it
```
Comment (rphmeier) [inline srml/example/src/lib.rs]: non-generic?

File: srml/support/src/dispatch.rs [hunk 2]
```diff
@@ -296,10 +298,26 @@ macro_rules! decl_module {
 		$system:ident;
 	) => {};
 
+	// None generic event
```
Comment (rphmeier) [inline srml/support/src/dispatch.rs]: non-generic?

**Comment by rphmeier** [comment]
seems fine except for typo or lingo I don't understand

**Comment by gavofyork** [comment]
@bkchr i guess this needs an according PR for polkadot's parachains module?

**Pull request merged**